packages/yarnpkg-core/sources/httpUtils.ts

```diff
@@ -92,7 +93,14 @@ export async function request(target: string, body: Body, {configuration, header
 
   for (const [glob, path] of configuration.get(`caFilePath`)) {
     if (micromatch.isMatch(url.hostname, glob)) {
-      extraHttpsOptions.certificateAuthority = await xfs.readFilePromise(path);
+      let entry = certCache.get(path);
+
+      if (!entry) {
+        entry = await xfs.readFilePromise(path);
+        certCache.set(path, entry);
+      }
+
+      extraHttpsOptions.certificateAuthority = await entry;
```

Comment:
```suggestion
      if (Buffer.isBuffer(entry) === false)
        entry = await entry;

      extraHttpsOptions.certificateAuthority = entry;
```
Avoids awaiting a already resolved promise